source/blog/2018-05-30-emberjs-accessibility-and-a11y-tools.md

```diff
@@ -0,0 +1,168 @@
+
+_This article was written as of Ember.js version 3.1. The same approach applies for all 2.x apps, with some changes to test syntax. Anyone can write a guest article --- get in touch if you are interested!_
+
+## Using ember-a11y-testing
+
+We don't have to learn a thousand rules to start building accessible apps. There are amazing community contributors who are working hard to make it as easy as possible to follow web best practices, and they created a set of tools referred to as [ember-a11y](https://github.com/ember-a11y).
+
+[ember-a11y-testing](https://github.com/ember-a11y/ember-a11y-testing) is an addon that reveals actionable improvements to your app's accessibility and guides us through making the changes.
+
+```bash
+ember install ember-a11y-testing
+ember server
+```
+
+After installing ember-a11y-testing, it gets to work right away! Navigate to [http://localhost:4200](http://localhost:4200) and take a look.
+
+![Striped elements showing contrast problems](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/a11y-testing-stripes.png)
+
+See these weird stripes? Those are an indication that we have some problems with the elements. Due to color choices and text size, the content might be hard for some people to read. The list of issues show up as a console error in this view too. But how do we fix them?
+
+![Instructions for resolving the error](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/console-errors.png)
+
+The link in the console error message brings us to this [color contrast checker](https://medium.com/r/?url=https%3A%2F%2Fdequeuniversity.com%2Frules%2Faxe%2F2.5%2Fcolor-contrast%3Fapplication%3DaxeAPI), where we can play around with font sizes, font colors, and backgrounds until the view meets the standards.
+
+In this way, learning the rules and standards of accessibility can be incremental. It's like having an accessibility expert inside the app. Once you have a handle on most of the issues, you can use some configuration settings to toggle the check on and off so that your development server build times are faster.
+
+### Add accessibility checks during Rendering tests
+
+[Rendering tests](https://guides.emberjs.com/v3.1.0/testing/#toc_rendering-tests) are a way to simulate user behavior, like clicking around and filling in forms. The ember-a11y-testing helpers can be added right in with your existing test suite. This test navigates to the index of the project and checks for any issues:
+
+```javascript
+import a11yAudit from 'ember-a11y-testing/test-support/audit';
+test('accessibility check', function(assert) {
```

Comment:
I'd add a comment here implying some irrelevant code is skipped. I imagine most people would know what the intent was, but some may not. I suggest something like:

```
import a11yAudit from 'ember-a11y-testing/test-support/audit';
// ...
test('accessibility check', function(assert) {
  visit('/');
  a11yAudit();
  andThen(() => assert.ok(true, 'no a11y errors found!'));
});
```

Reply:
done